src/sgame/sg_utils.cpp

```diff
@@ -296,7 +298,7 @@ void G_TeleportPlayer( gentity_t *player, glm::vec3 const& origin, glm::vec3 con
 	BG_PlayerStateToEntityState( &player->client->ps, &player->s, true );
 
 	// use the precise origin for linking
-	VectorCopy( player->client->ps.origin, player->r.currentOrigin );
+	G_SetOrigin( player, origin );
```

Comment:
I tested and this part wasn't necessary to prevent the bug. G_SetOrigin sets the `s.pos` stuff which I believe is not used by clients.

Reply:
I figured it's more correct to use a proper named helper function than not? The intent is more clear using this function than a rando vector copy.

Reply:
Using G_SetOrigin sets a bad example because this function doesn't actually work for clients. The actual load-bearing part of this function is `VectorCopy( origin, player->client->ps.origin );` followed by `BG_PlayerStateToEntityState( &player->client->ps, &player->s, true );`. If you delete the VectorCopy, it doesn't work at all. If you delete the G_SetOrigin, then the position for linking is just wrong for 1 frame until it gets updated by pmove.

Reply:
https://github.com/Unvanquished/Unvanquished/blob/master/src/sgame/sg_client.cpp#L1674

We use it for spawning as well?

Reply:
It's the same situation - G_SetOrigin doesn't do nearly enough on its own. There are 3 other places in ClientSpawn where some origin is set. And the only line we actually want from it is the one to set currentOrigin.

Reply:
done